# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineWebRTCAudio.cpp
# Commit: abe51ad662bc
# Full Hash: abe51ad662bc6efbe8c637c75cd2630788d49446
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Update logging for MediaEngineWebRTCMicrophoneSource et al. r=padenot
#   
#   This patch adds identifiers to existing log messages in the mic source and
#   AudioInputProcessing, and adds new log messages for complete tracing of frames.
#   
# ==============================================================================

diff -r 0ae30eae800d -r abe51ad662bc dom/media/webrtc/MediaEngineWebRTCAudio.cpp
--- a/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Thu Nov 05 16:44:00 2020 +0000
+++ b/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Thu Nov 05 16:44:38 2020 +0000
@@ -124,8 +124,8 @@
   prefs.mChannels = c.mChannelCount.Get(std::min(prefs.mChannels, maxChannels));
   prefs.mChannels = std::max(1, std::min(prefs.mChannels, maxChannels));
 
-  LOG("Audio config: aec: %d, agc: %d, noise: %d, channels: %d",
-      prefs.mAecOn ? prefs.mAec : -1, prefs.mAgcOn ? prefs.mAgc : -1,
+  LOG("Mic source %p Audio config: aec: %d, agc: %d, noise: %d, channels: %d",
+      this, prefs.mAecOn ? prefs.mAec : -1, prefs.mAgcOn ? prefs.mAgc : -1,
       prefs.mNoiseOn ? prefs.mNoise : -1, prefs.mChannels);
 
   *aOutPrefs = prefs;
@@ -484,7 +484,8 @@
   MOZ_ASSERT(mState != kStarted, "Source not stopped");
 
   mState = kReleased;
-  LOG("Audio device %s deallocated", NS_ConvertUTF16toUTF8(mDeviceName).get());
+  LOG("Mic source %p Audio device %s deallocated", this,
+      NS_ConvertUTF16toUTF8(mDeviceName).get());
   return NS_OK;
 }
 
@@ -508,7 +509,8 @@
         track->Resume();  // Suspended by MediaManager
       }));
 
-  LOG("Track %p registered for microphone capture", aTrack.get());
+  LOG("Mic source %p Track %p registered for microphone capture", this,
+      aTrack.get());
 }
 
 class StartStopMessage : public ControlMessage {
@@ -654,6 +656,10 @@
   MOZ_ASSERT(aGraph->OnGraphThread());
   if (!mSkipProcessing && aPassThrough && mPacketizerInput) {
     MOZ_ASSERT(mPacketizerInput->PacketsAvailable() == 0);
+    LOG_FRAME(
+        "AudioInputProcessing %p Appending %u frames of null data for data "
+        "discarded in the packetizer",
+        this, mPacketizerInput->FramesAvailable());
     mSegment.AppendNullData(mPacketizerInput->FramesAvailable());
     mPacketizerInput->Clear();
   }
@@ -693,8 +699,10 @@
     if (aLevel != EchoCancellation::SuppressionLevel::kLowSuppression &&
         aLevel != EchoCancellation::SuppressionLevel::kModerateSuppression &&
         aLevel != EchoCancellation::SuppressionLevel::kHighSuppression) {
-      LOG_ERROR("Attempt to set invalid AEC suppression level %d",
-                static_cast<int>(aLevel));
+      LOG_ERROR(
+          "AudioInputProcessing %p Attempt to set invalid AEC suppression "
+          "level %d",
+          this, static_cast<int>(aLevel));
 
       aLevel = EchoCancellation::SuppressionLevel::kModerateSuppression;
     }
@@ -711,14 +719,18 @@
   if (aMode != GainControl::Mode::kAdaptiveAnalog &&
       aMode != GainControl::Mode::kAdaptiveDigital &&
       aMode != GainControl::Mode::kFixedDigital) {
-    LOG_ERROR("Attempt to set invalid AGC mode %d", static_cast<int>(aMode));
+    LOG_ERROR("AudioInputProcessing %p Attempt to set invalid AGC mode %d",
+              this, static_cast<int>(aMode));
 
     aMode = GainControl::Mode::kAdaptiveDigital;
   }
 
 #if defined(WEBRTC_IOS) || defined(ATA) || defined(WEBRTC_ANDROID)
   if (aMode == GainControl::Mode::kAdaptiveAnalog) {
-    LOG_ERROR("Invalid AGC mode kAgcAdaptiveAnalog on mobile");
+    LOG_ERROR(
+        "AudioInputProcessing %p Invalid AGC mode kAgcAdaptiveAnalog on "
+        "mobile",
+        this);
     MOZ_ASSERT_UNREACHABLE(
         "Bad pref set in all.js or in about:config"
         " for the auto gain, on mobile.");
@@ -739,8 +751,10 @@
       aLevel != NoiseSuppression::Level::kModerate &&
       aLevel != NoiseSuppression::Level::kHigh &&
       aLevel != NoiseSuppression::Level::kVeryHigh) {
-    LOG_ERROR("Attempt to set invalid noise suppression level %d",
-              static_cast<int>(aLevel));
+    LOG_ERROR(
+        "AudioInputProcessing %p Attempt to set invalid noise suppression "
+        "level %d",
+        this, static_cast<int>(aLevel));
 
     aLevel = NoiseSuppression::Level::kModerate;
   }
@@ -813,6 +827,9 @@
       MOZ_ASSERT(mPacketizerInput);
       MOZ_ASSERT((buffering - mLiveBufferingAppended) ==
                  mPacketizerInput->mPacketSize);
+      LOG_FRAME("AudioInputProcessing %p Inserting %" PRId64
+                " frames of silence due to buffer increase",
+                this, buffering - mLiveBufferingAppended);
       mSegment.InsertNullDataAtStart(buffering - mLiveBufferingAppended);
       mLiveBufferingAppended = buffering;
     } else if (MOZ_UNLIKELY(buffering < mLiveBufferingAppended)) {
@@ -823,6 +840,9 @@
                  (mLiveBufferingAppended - buffering));
       TrackTime frames =
           std::min(mSegment.GetDuration(), mLiveBufferingAppended - buffering);
+      LOG_FRAME("AudioInputProcessing %p Removing %" PRId64
+                " frames of silence due to buffer decrease",
+                this, frames);
       mLiveBufferingAppended -= frames;
       mSegment.RemoveLeading(frames);
     }
@@ -832,14 +852,18 @@
     if (!mLiveFramesAppended) {
       // First real data being pulled in. Add the appropriate amount of
       // buffering before the real data to avoid glitches.
-      LOG_FRAME("Buffering %" PRId64 " frames of pre-silence for %u channels.",
-                buffering, mRequestedInputChannelCount);
+      LOG_FRAME("AudioInputProcessing %p Buffering %" PRId64
+                " frames of pre-silence for %u channels.",
+                this, buffering, mRequestedInputChannelCount);
       mSegment.InsertNullDataAtStart(buffering - mLiveBufferingAppended);
       mLiveFramesAppended = true;
       mLiveBufferingAppended = buffering;
     }
     MOZ_ASSERT(buffering == mLiveBufferingAppended);
     TrackTime frames = std::min(mSegment.GetDuration(), delta);
+    LOG_FRAME("AudioInputProcessing %p Appending %" PRId64
+              " frames of real data for %u channels.",
+              this, frames, mRequestedInputChannelCount);
     aSegment->AppendSlice(mSegment, 0, frames);
     mSegment.RemoveLeading(frames);
     delta -= frames;
@@ -855,8 +879,9 @@
     return;
   }
 
-  LOG_FRAME("Pulling %" PRId64 " frames of silence for %u channels.", delta,
-            mRequestedInputChannelCount);
+  LOG_FRAME("AudioInputProcessing %p Pulling %" PRId64
+            " frames of silence for %u channels.",
+            this, delta, mRequestedInputChannelCount);
 
   // This assertion fails if we append silence here after having appended live
   // frames. Before appending live frames we should add sufficient buffering to
@@ -973,6 +998,9 @@
         aRate / 100, aChannels);
   }
 
+  LOG_FRAME("AudioInputProcessing %p Appending %zu frames to packetizer", this,
+            aFrames);
+
   // Packetize our input data into 10ms chunks, deinterleave into planar channel
   // buffers, process, and append to the right MediaStreamTrack.
   mPacketizerInput->Input(aBuffer, static_cast<uint32_t>(aFrames));
@@ -1066,8 +1094,8 @@
       continue;
     }
 
-    LOG_FRAME("Appending %" PRIu32 " frames of packetized audio",
-              mPacketizerInput->mPacketSize);
+    LOG_FRAME("AudioInputProcessing %p Appending %u frames of packetized audio",
+              this, mPacketizerInput->mPacketSize);
 
     // We already have planar audio data of the right format. Insert into the
     // MTG.
@@ -1112,7 +1140,8 @@
                                  write_channels.Elements());
   }
 
-  LOG_FRAME("Appending %zu frames of raw audio", aFrames);
+  LOG_FRAME("AudioInputProcessing %p Appending %zu frames of raw audio", this,
+            aFrames);
 
   MOZ_ASSERT(aChannels == channels.Length());
   mSegment.AppendFrames(buffer.forget(), channels, aFrames, mPrincipal);
